pandas/_testing/_warnings.py

```diff
@@ -130,7 +130,8 @@ def _assert_caught_expected_warning(
             saw_warning = True
 
             if check_stacklevel and issubclass(
-                actual_warning.category, (FutureWarning, DeprecationWarning)
+                actual_warning.category,
+                (FutureWarning, DeprecationWarning, UserWarning),
```

Comment:
the test which checked the warning passes on `main` because stacklevel is only checked for `FutureWarning` and `DeprecationWarning`